### PR TITLE
Improve linearSCan warning and make it off by default #1555

### PR DESCRIPTION
**Pull request by peter-lawrey**

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/queue/impl/single/SCQIndexing.java [hunk 1]
```diff
@@ -348,25 +349,30 @@ private ScanResult linearScan(@NotNull final Wire wire,
...
-        }
+        if (end < start + 100_000)
+            return;
+
+        doPrintLinerScanTime(toIndex, fromKnownIndex, start, desc, end);
     }
 
-    private boolean printLinearScanTime(long toIndex, long fromKnownIndex, long start, long end,
-                                        String desc) {
+    private void doPrintLinerScanTime(long toIndex, long fromKnownIndex, long start, String desc, long end) {
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/queue/impl/single/SCQIndexing.java]: Liner should be Linear

File: src/main/java/net/openhft/chronicle/queue/impl/single/SCQIndexing.java [hunk 2]
```diff
@@ -79,6 +79,7 @@ class SCQIndexing extends AbstractCloseable implements Indexing, Demarshallable,
     int linearScanCount;
     int linearScanByPositionCount;
     Collection<Closeable> closeables = new ArrayList<>();
+    private long lastScanedIndex = -1;
```
Comment (tgd) [inline src/main/java/net/openhft/chronicle/queue/impl/single/SCQIndexing.java]: `lastScanedIndex` -> `lastScannedIndex`

**Review by tgd — CHANGES_REQUESTED** [review]
Thanks Peter - just two minor spelling fixes needed

File: src/main/java/net/openhft/chronicle/queue/impl/single/SCQIndexing.java [hunk 3]
```diff
@@ -348,25 +349,30 @@ private ScanResult linearScan(@NotNull final Wire wire,
...
+        // still warming up?
+        if (toIndex <= 1)
+            return;
+
+        // took too long to scan?
         long end = System.nanoTime();
-        if (end > start + 100_000) {
-            printLinearScanTime(toIndex, fromKnownIndex, start, end, "linearScan by index");
-        }
+        if (end < start + 100_000)
```
Comment (nicktindall) [inline src/main/java/net/openhft/chronicle/queue/impl/single/SCQIndexing.java]: Should `100_000` be a constant?

perhaps

```java
private static final int LINEAR_SCAN_WARN_THRESHOLD_NS = TimeUnit.MICROSECONDS.toNanos(100);
```

**Review by tgd — APPROVED** [review]

**Pull request merged**